runners/java-fn-execution/src/test/java/org/apache/beam/runners/fnexecution/graph/LengthPrefixUnknownCodersTest.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.beam.runners.fnexecution.graph;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.IOException;
+import org.apache.beam.model.pipeline.v1.RunnerApi.Components;
+import org.apache.beam.model.pipeline.v1.RunnerApi.MessageWithComponents;
+import org.apache.beam.runners.core.construction.CoderTranslation;
+import org.apache.beam.runners.core.construction.RehydratedComponents;
+import org.apache.beam.sdk.coders.ByteArrayCoder;
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.coders.KvCoder;
+import org.apache.beam.sdk.coders.LengthPrefixCoder;
+import org.apache.beam.sdk.coders.StringUtf8Coder;
+import org.apache.beam.sdk.coders.VarIntCoder;
+import org.apache.beam.sdk.transforms.windowing.GlobalWindow;
+import org.apache.beam.sdk.util.WindowedValue;
+import org.apache.beam.sdk.values.KV;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests for {@link LengthPrefixUnknownCoders}. */
+@RunWith(JUnit4.class)
+public class LengthPrefixUnknownCodersTest {
+  private static final Coder<WindowedValue<KV<String, Integer>>> windowedValueCoder =
+      WindowedValue.getFullCoder(KvCoder.of(StringUtf8Coder.of(), VarIntCoder.of()),
+          GlobalWindow.Coder.INSTANCE);
+
+  private static final Coder<WindowedValue<KV<String, Integer>>> prefixedWindowedValueCoder =
+      WindowedValue.getFullCoder(KvCoder.of(LengthPrefixCoder.of(StringUtf8Coder.of()),
+          LengthPrefixCoder.of(VarIntCoder.of())), GlobalWindow.Coder.INSTANCE);
+
+  private static final Coder<WindowedValue<KV<byte[], byte[]>>>
+      prefixedAndReplacedWindowedValueCoder = WindowedValue.getFullCoder(
+      KvCoder.of(
+          LengthPrefixCoder.of(ByteArrayCoder.of()),
+          LengthPrefixCoder.of(ByteArrayCoder.of())),
+      GlobalWindow.Coder.INSTANCE);
+
+  /** Test wrapping unknown coders with {@code LengthPrefixCoder}. */
```

Comment:
I never remember if UTF-8 Strings are well known or not; maybe consider a `CustomCoder` just for the obviousness of it being unknown?

Reply:
Done